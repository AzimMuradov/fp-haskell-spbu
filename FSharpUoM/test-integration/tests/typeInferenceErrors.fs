let x = 2; 
let x = 10;; // Right: "Duplicate definition of value 'x'"

let x = 2;
let x y = y * 10;; // Right: "Duplicate definition of value 'x'"

[<Measure>] type m;
[<Measure>] type m;; // Right: "Duplicate definition of measure type 'm'"

let f x = y + x;; // Right: "Unbound variable 'y'"

let f x = f (x - 1);; // Right: "Unbound variable 'f'"

1<m> + 1<cm>;; // Right: "Measure 'm' do not define"

1 && (false || true);; // Right: "It is not possible to apply this operation between 'int' and 'bool'"

1 + 2.9;; // Right: "The type 'int' does not match the type 'double'"

[<Measure>] type m;
1.0<m> + 1<m>;; // Right: "The type 'double<m>' does not match the type 'int<m>'"

[<Measure>] type m;
[<Measure>] type cm;
1<m> + 1<cm>;; // Right: "The type 'm' does not match the type 'cm'"

[<Measure>] type m;
[<Measure>] type cm;
1<m> * 1<cm> + 1<m>;; // Right: "The type 'cm m' does not match the type 'm'"