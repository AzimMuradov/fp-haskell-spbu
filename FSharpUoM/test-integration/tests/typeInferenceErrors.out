Duplicate definition of value 'x'
Duplicate definition of value 'x'
Duplicate definition of measure type 'm'
Unbound variable 'y'
Unbound variable 'f'
Measure 'm' do not define
It is not possible to apply this operation between 'int' and 'bool'
The type 'int' does not match the type 'double'
The type 'double<m>' does not match the type 'int<m>'
The type 'm' does not match the type 'cm'
The type 'cm m' does not match the type 'm'
